File: test-suite/src/test/java/io/micronaut/docs/client/ApiClientListenerTest.java
package io.micronaut.docs.client;

import io.kubernetes.client.openapi.ApiClient;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
class ApiClientListenerTest {

    @Inject
    ApiClient apiClient;

    @Test
    void testTheListenerCustomizesTheOkHttpClient() {
        assertEquals(5345, apiClient.getHttpClient().readTimeoutMillis());
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/client/ApiClientListenerTest.kt
package io.micronaut.docs.client

import io.kubernetes.client.openapi.ApiClient
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautTest
class ApiClientListenerTest {

    @Inject
    lateinit var apiClient: ApiClient

    @Test
    fun testTheListenerCustomizesTheOkHttpClient() {
        assertEquals(5345, apiClient.httpClient.readTimeoutMillis)
    }
}
